sync/src/poison.rs

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) 2023 The MobileCoin Foundation
```

Comment:
I'm up for suggestions on the copyright, not sure it's appropriate when this is mostly copied from https://github.com/rust-lang/rust/blob/master/library/std/src/sync/poison.rs

Reply:
I would put the Rust Foundation copyright either above or below ours...

Reply:
Added rust foundation to the file, it's not clear who actually is the copyright holder due to, https://github.com/rust-lang/rust/blob/master/COPYRIGHT
> Copyrights in the Rust project are retained by their contributors. No copyright assignment is required to contribute to the Rust project.